docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+                placeholder="User Name", name="name", required=True
+            ),
+            rx.input(
+                placeholder="user@reflex.dev",
+                name="email",
+            ),
+            rx.select(
+                ["Male", "Female"],
+                placeholder="Male",
+                name="gender",
+            ),
+            rx.button("Submit", type="submit"),
+        ),
+        on_submit=State3.add_user,
+        reset_on_submit=True,
+    )
+```
+
+```python eval
+rx.vstack(
+    form(),
+    rx.table.root(
+        rx.table.header(
+            rx.table.row(
+                rx.table.column_header_cell("Name"),
+                rx.table.column_header_cell("Email"),
+                rx.table.column_header_cell("Gender"),
+            ),
+        ),
+        rx.table.body(
+            rx.foreach(
+                State3.users, show_user
+            ),
+        ),
+        variant="surface",
+        size="3",
+    ),
+)
+```
+
+```python
+class State(rx.State):
+    users: list[User] = [
+        User(name="Danilo Sousa", email="danilo@example.com", gender="Male"),
+        User(name="Zahra Ambessa", email="zahra@example.com", gender="Female"),
+    ]
+
+    def add_user(self, form_data: dict):
+        self.users.append(User(**form_data))
+
+
+def show_user(user: User):
+    """Show a person in a table row."""
+    return rx.table.row(
+        rx.table.cell(user.name),
+        rx.table.cell(user.email),
+        rx.table.cell(user.gender),
+    )
+
+def form():
+    return rx.form(
+        rx.vstack(
+            rx.input(
+                placeholder="User Name", name="name", required=True
+            ),
+            rx.input(
+                placeholder="user@reflex.dev",
+                name="email",
+            ),
+            rx.select(
+                ["Male", "Female"],
+                placeholder="Male",
+                name="gender",
+            ),
+            rx.button("Submit", type="submit"),
+        ),
+        on_submit=State.add_user,
+        reset_on_submit=True,
+    )
+
+def index() -> rx.Component:
+    return rx.vstack(
+        form(),
+        rx.table.root(
+            rx.table.header(
+                rx.table.row(
+                    rx.table.column_header_cell("Name"),
+                    rx.table.column_header_cell("Email"),
+                    rx.table.column_header_cell("Gender"),
+                ),
+            ),
+            rx.table.body(
+                rx.foreach(
+                    State.users, show_user
+                ),
+            ),
+            variant="surface",
+            size="3",
+        ),
+    )
+```
+
+
+### Putting the Form in an Overlay
+
+In Reflex we like to make the user interaction as intuitive as possible. Placing the form we just constructed in an overlay creates a focused interaction by dimming the background, and ensures a cleaner layout when you have multiple action points such as editing and deleting as well.
```

Comment:
What.... is an overlay? Why do I want this? You kind of describe it, but ehh (maybe I have low attention span)

I think in general, maybe show the product of this component first. Then as user I'm like "hey, that's cool! how can I add that?". Then say that it's an overlay and show how to use it.

Otherwise it's like "huh. why do I need this? what is this?". By this point I might tune out. If I don't, by the end "ohhh that's what it does. ok sure that's cool".

Also add a comma after "In Reflex".

Reply:
I think we see the use of the overlay in the original demo at the top of the page